File: app/src/main/java/com/vhn/SurfaceDuoUtils.java
package com.vhn;

import android.content.pm.PackageManager;

public class SurfaceDuoUtils {
    private static int isDeviceSurfaceDuoComputed = -1;
    static final String feature = "com.microsoft.device.display.displaymask";

    public static boolean isDeviceSurfaceDuo(PackageManager pm) {
        if (isDeviceSurfaceDuoComputed == -1)
            isDeviceSurfaceDuoComputed = pm.hasSystemFeature(feature) ? 1 : 0;
        return isDeviceSurfaceDuoComputed == 1;
    }
}

File: app/src/main/java/com/vhn/SurfaceDuoUtils.kt
package com.vhn

import android.content.pm.PackageManager

object SurfaceDuoUtils {
    private var isDeviceSurfaceDuoComputed = -1
    const val feature = "com.microsoft.device.display.displaymask"
    fun isDeviceSurfaceDuo(pm: PackageManager): Boolean {
        if (isDeviceSurfaceDuoComputed == -1) isDeviceSurfaceDuoComputed = if (pm.hasSystemFeature(
                feature)
        ) 1 else 0
        return isDeviceSurfaceDuoComputed == 1
    }
}